pkg/extractor/extractor.go

```diff
@@ -83,22 +91,55 @@ func extractYamlConfigurations(content string) (*[]Configuration, error) {
 
 	var err error
 	for {
-		var doc = map[string]interface{}{}
-		err = yamlDecoder.Decode(&doc)
+		var anyTypeStore interface{}
+		err = yamlDecoder.Decode(&anyTypeStore)
+		if err != nil {
+			if err == io.EOF {
+				break
+			} else {
+				return nil, err
+			}
+		}
+
+		yamlByteArray, err := yaml.Marshal(anyTypeStore)
 		if err != nil {
-			break
+			return nil, err
 		}
 
-		if len(doc) > 0 {
-			configurations = append(configurations, doc)
+		jsonByte, err := yamlConvertor.YAMLToJSON(yamlByteArray)
+		if err != nil {
+			return nil, err
+		}
+
+		configurations = append(configurations, extractConfigurationK8sData(jsonByte))
+	}
+
+	return &configurations, nil
+}
+
+func extractConfigurationK8sData(content []byte) Configuration {
+	var configuration Configuration
+	var jsonObject map[string]interface{}
+	configuration.Payload = content
```

Comment:
is Payload the original content (string of the yaml)?
maybe you can change it to a name that explains better what that is like `rawJson`?